lib/Sema/MiscDiagnostics.cpp

```diff
@@ -3710,12 +3710,22 @@ static void diagnoseUnintendedOptionalBehavior(TypeChecker &TC, const Expr *E,
     }
 
     void emitSilenceOptionalAnyWarningWithCoercion(Expr *E, Type destType) {
+      Diag<Type, StringRef> silenceDiag = diag::silence_optional_to_any;
+
+      if (auto declRefExpr = dyn_cast<DeclRefExpr>(E)) {
```

Comment:
You should use [`hasImplicitlyUnwrappedResult`](https://github.com/apple/swift/blob/6b2821c52548b0050b59382885e1a73361051adb/lib/Sema/MiscDiagnostics.cpp#L3726-L3757) which I added in https://github.com/apple/swift/pull/16434.

I handles function calls, subscripts, and member references.